### PR TITLE
Proposal to rework the PR template

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

WIP

### ❓ Context

- **Impacted projects**: `` 
- **Linked resource(s)**: [] 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Comment by haammar-ledger** [comment]
> _Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready._

I would separate as 2 different bullet items, and I would add another one for _"Make sure a changeset is included using `pnpm changeset`"_
(From experience, it seems few people notice the automatic comment about changeset below the description...)

> _The impacts are:_

Seems a bit vague, what is expected here? Can you share examples?

> _The PR has been tested thoroughly_

I would rephrase this as _"The PR has been thoroughly covered with tests"_, to avoid confusion with manual testing (which the reviewer doesn't really know about).

> _Any new dependencies have been justified and documented._

I would add _"and compatibility has been verified on desktop, iOS and Android"_

**Comment by gre** [comment]
>> Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.
> 
> I would separate as 2 different bullet items

@haammar-ledger  in my mind, this was just an informative one liner. kinda a reminder that PR creator need to watch CI and then ping the reviewers on github directly (or whatever other ways), it felt a bit doublon to consider checkbox.

- good point for changeset. i think we will also refine them on the danger.js iteration
- indeed. i need to see how to make explicit about the manual vs automated tests.
- compatibility has been verified on desktop, iOS and Android
  - to me this is already totally assessed by e2e automation today. we no longer have the history problem.
  - however, i think it could be interesting more generally to say that a dep needs to not be big for the bundle size and be web compatible when it's in a library 🤔  but honestly very hard to try to do a oneliner with all info. i think some of these topics will go back into CONTRIBUTING.md when we need more explanations

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+Please make sure to read CONTRIBUTING.md if you have not already. Pull Requests that do not comply with the rules will be arbitrarily closed.
 -->
 
 ### 📝 Description
 
-_Replace this text by a clear and concise description of what this pull request is about and why it is needed._
+_Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
+_For visual features, please attach screenshots or video recordings to demonstrate the changes._
+_For libraries, you can add a code sample of how to use it._
+_For bugfixes, you can explain the previous behavior and how it was fixed._
```
Comment (chabroA) [inline .github/pull_request_template.md:11]: Could be worthwhile to add some templates as (hidden) comments to easily / consistently display visual changes.

Examples of PRs descriptions where these have been used in the past:
- For images: https://github.com/LedgerHQ/ledger-live/pull/5069
- For videos: https://github.com/LedgerHQ/ledger-live/pull/4790 

```suggestion
_For bugfixes, you can explain the previous behavior and how it was fixed._

<!--
You can use the templates bellow to clearly display visual changes:

For videos

| Before  | After |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/13440061/129455220-23fa27a2-c8f0-4953-b291-b4893959d5d9.mp4">  | <video src="https://user-images.githubusercontent.com/13440061/129455220-23fa27a2-c8f0-4953-b291-b4893959d5d9.mp4">|

For images:

| Before | After |
|--------|--------|
| ![before](https://github.com/LedgerHQ/ledger-live/assets/9203826/026c13c0-540c-486c-b5f7-b3858c153933) | ![after](https://github.com/LedgerHQ/ledger-live/assets/9203826/aa9666d6-f295-41ae-8d05-c6690c455423) | 

-->

```

**Review by chabroA — COMMENTED** [review]
Some proposed improvements

File: .github/pull_request_template.md (hunk 1 above)
Comment (gre) [reply .github/pull_request_template.md:11]: would add too much verbosity in the textarea i think, i rarely think we will need to do these before after images

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-- **Impacted projects**: `` <!-- Add the list of end user projects impacted by the change inside of the ` `, like so `LLM, LLD`. -->
-- **Linked resource(s)**: [] <!-- Attach any ticket number if relevant inside the brackets, like so [LIVE-0000]. (JIRA / Github issue number) -->
+- **JIRA or GitHub issue**: <!-- Attach the relevant ticket number if applicable. (e.g., JIRA-123 or #123) -->
 
 ### ✅ Checklist
 
-- [ ] **Test coverage** <!-- Are your changes covered by tests? Features must be tested, bugfixes must include a test that would have detected the issue. -->
-- [ ] **Atomic delivery** <!-- Is this pull request standalone? In order words, does it depend on nothing else? Please explain if not checked. -->
-- [ ] **No breaking changes** <!-- If there are breaking changes, please explain why. -->
+- Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.
```
[the remark this answers is not in the document]
Comment (gre) [reply .github/pull_request_template.md:19]: good point

**Comment by mohammed-ledger** [comment]
Looks great, thanks @gre. Do we also need to update `tools/github-bot/src/features/autoClose/tools.ts` `isValidBody(..)`?

**Comment by valpinkman** [comment]
> Looks great, thanks @gre. Do we also need to update `tools/github-bot/src/features/autoClose/tools.ts` `isValidBody(..)`?

Very valid point. I also told @gre about i. I will check that the parser is still working 👍

File: .github/pull_request_template.md (hunk 1 above)
Comment (chabroA) [reply .github/pull_request_template.md:11]: Since it's in comments I don't really see the drawbacks. On the contrary, if you need it, you uncomment it and / or copy paste, if you don't, you just keep it as comment.
But up to you in the end

**Comment by gre** [comment]
@valpinkman to me looking at the code it works
